### PR TITLE
Add ApplicationAction#routes to access Hanami app routes

### PR DESCRIPTION
**Pull request by waiting-for-dev**
References hanami/hanami#1119

### CODE REVIEW

File: spec/integration/hanami/controller/application_action/routes_spec.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+          end
+        end
+      RUBY
+
+      write "slices/main/actions/test_action.rb", <<~RUBY
+        module Main
+          module Actions
+            class TestAction < Hanami::Action
+              def handle(req, res)
+                res.body = "Hello, world!"
```
Comment (jodosha) [inline spec/integration/hanami/controller/application_action/routes_spec.rb:35]: @waiting-for-dev I would take the chance of testing the fact that those helpers are accessible here.

```suggestion
                res.body = routes.root_url
```

File: spec/integration/hanami/controller/application_action/routes_spec.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            end
+          end
+        end
+      RUBY
+
+      require "hanami/init"
+
+      action = Main::Slice["actions.test_action"]
+
+      expect(action.routes.path(:root)).to eq("/")
```
Comment (jodosha) [inline spec/integration/hanami/controller/application_action/routes_spec.rb:46]: @waiting-for-dev I would perform a request and assert that the body of the response is equal to the root URL.

**Review by jodosha — CHANGES_REQUESTED** [review]
@waiting-for-dev Thanks for this PR.

I believe we should use Dependency Injection here.
Preferred way is to import via `dry-system` (`include Deps["routes"]`).

**Comment by waiting-for-dev** [comment]
Closing in favor of #358

**Pull request closed without merging**